Home redesign, reading navigation updates, and progress reset

Cipher73 opened the pull request:
<img width="720" height="1600" alt="image" src="https://github.com/user-attachments/assets/6bd03dd3-05ba-4e30-8129-d55a446a8899" />
<img width="720" height="1600" alt="image" src="https://github.com/user-attachments/assets/0358ea1e-defb-478b-9752-f13616500866" />


- ensure that all the days before first lauch date are marked as not read instead of missed
- added reset progress functionality and card on the settings page
- added pop up for settings and share buttons on the reading pages
- added home button on the reading pages
- added functionality to swipe left and right on reading page

Cipher73 committed:
- redesign on the home screen

jeremyscodes committed:
fix visuals

Cipher73 committed:
share text formating

Cipher73 committed:
fix lint issues

Cipher73 committed:
add home button to app bar on preface_forward screen

jeremyscodes reviewed (commented):
There are quite a few commented out blocks of code we can delete.
Also there are some unnecessary comments like 'reverted back to original...'
Otherwise LGTM

Cipher73 commented:
Thank you, I will do a cleanup with the next pr , or should i do it here?

jeremyscodes committed:
adds fancy divider to reading page

jeremyscodes committed:
DropText initial letter

jeremyscodes committed:
DropText initial letter

jeremyscodes committed:
DropText initial letter

jeremyscodes committed:
DropText initial letter

jeremyscodes committed:
improves sizing

Cipher73 committed:
Merge branch 'develop' of https://github.com/Family-Altar/family-altar-mobile into develop

Cipher73 committed:
Merge branch 'develop' into home_screen_redesign_AN

jeremyscodes commented:
> Thank you, I will do a cleanup with the next pr , or should i do it here?

As mentioned on Whatsapp, might as well do it here

jeremyscodes committed:
no longer remove new lines after updating .txt

jeremyscodes committed:
lint fixes

jeremyscodes committed:
MERGE FIX

Cipher73 committed:
- ensure that all the days before first lauch date are marked as not read instead of missed
- added reset progress functionality and card on the settings page
- added pop up for settings and share buttons on the reading pages
- added home button on the reading pages
- added functionality to swipe left and right on reading page

Cipher73 committed:
fix for lint errors

Cipher73 committed:
code cleanup

Cipher73 committed:
test commint

jeremyscodes commented on lib/main.dart:
```diff
@@ -64,6 +67,9 @@ final GoRouter _router = GoRouter(
   WidgetsFlutterBinding.ensureInitialized();
 
+  // Initialize first launch date (app download date) - only sets it once
+  await ReadingBloc.initializeFirstLaunchDate();
```
This will run when the app is started after force closed. It will over-write the date each time

Cipher73 replied:
@jeremyscodes  i double checked the function since it checks if _firstLaunchKey is in persistent storage, it will not overwite it , to be on the same side i have added an extra check 

- first check if the first launch date is set in the shared preferences
- also check if there is no reading entries in the shared preferences

Cipher73 committed:
added an extra check for existing reading entries before setting the first launch date in shared preferences. also uncomment some code for setting a gradeitn on the book selection screen

Cipher73 committed:
- Merge branch 'develop' into home_screen_redesign_AN
- more code cleanup

Cipher73 committed:
code clean up

Cipher73 committed:
Refactor ReaderScreen layout for improved readability and structure

Cipher73 committed:
Fix typo in reader_screen.dart for divider color property

Cipher73 committed:
Refactor MissedDaysScreen layout and improve code readability; remove unused imports and adjust text styling

Cipher73 committed:
Update text formatting in MissedDaysScreen for improved readability

jeremyscodes committed:
adds back correct \n processing
